lib/krikri/mapper.rb

```diff
@@ -90,23 +90,55 @@ def mapping_exception_desc(rec)
     ##
     # A SoftwareAgent to run mapping processes.
     #
+    # @example
+    #
+    #   To map the records harvested by the harvest activity with ID 1:
+    #
+    #   Krikri::Mapper::Agent.enqueue(
+    #     :mapping,
+    #     opts = {
+    #       name: 'scdl_qdc',
+    #       generator_uri: 'http://ldp.local.dp.la/ldp/activity/1'
+    #     }
+    #   )
+    #   Or, supposing you already have a harvest Activity instance named
+    #   `my_activity':
+    #   [ ... ]
+    #     opts = {
+    #       name: 'scdl_qdc',
+    #       activity: my_activity
+    #     }
+    #
     # @see: Krikri::SoftwareAgent, Krikri::Activity
     class Agent
       include SoftwareAgent
 
-      attr_reader :name, :generator_uri
+      attr_reader :name
 
       def self.queue_name
         :mapping
       end
 
+      def self.entity_behavior
+        Krikri::AggregationEntityBehavior
+      end
+
+      def self.generator_entity_behavior
```

Comment:
Make these instance methods. Otherwise you need to subclass the `Agent` to get a different behavior, which seems to defeat the purpose of injecting them.